### ISSUE TITLE
C++/WinRT

### ISSUE THREAD

**Issue by Gavin-Williams** [microsoft/DirectX-Graphics-Samples#723]
For a while now, Microsoft have been saying to use C++/WinRT instead of C++/CX. Why then, are these samples still using C++/CX?

When will we get good resources explaining and showing how to use C++/WinRT and DirectX together? Do they exist already and I just can't find them?

**Comment by walbourn** [comment]
The use of C++/CX vs. C++/WinRT is pretty much orthogonal to the use of DirectX. The samples use C++/CX to establish the UWP window and message loop, but the bulk of the samples are pure C++ using DirectX via COM.

If you want some examples of a C++/WinRT 'basic window message loop', take a look at [directx-vs-templates](https://github.com/walbourn/directx-vs-templates).

**Comment by Gavin-Williams** [comment]
@walbourn Thanks for that clarification. I would like to mention though, that it's confusing and presents a barrier when official samples use technology that is either deprecated or not-recommended, and in this case, not even supported. It's hard enough working this stuff out given that there is a combination of complex technologies: C++, DirectX and C++/WinRT, without throwing in the extra difficulty that the code you present as official examples does not even work in the current application development environment. Why would you leave it the way it is? 

It just seems like there is something lacking in the presentation of graphics technologies over at Microsoft. No one writing a DirectX tutorial today would use C++/CX. Updating this example code should be on your timeline I would think.

**Comment by walbourn** [comment]
If you could, please rename the issue to **Convert UWP samples to use C++/WinRT instead of C++/CX**

**Comment by MRahilly** [comment]
The samples in Petzold's PW6 which are largely UWP should be translated to C++/WinRT.

**Comment by walbourn** [comment]
> The samples in Petzold's PW6 which are largely UWP should be translated to C++/WinRT.

To be fair, that book was written for Windows 8 which predates C++/WinRT projections.

**Comment by MRahilly** [comment]
> To be fair, that book was written for Windows 8 which predates C++/WinRT projections.
I find that the samples work well when translated to C++/WInRT. A number of Windows 8 API's have been replaced but this is well documented. The biggest problem is trying to figure out C++CX async task<> calls and how to replace things like C++/CX CancellationTokenSource which came across from .NET
I translated the Chapter03 Basic Event Handling samples from C++/CX to C++/WinRT and then to the Rust Crate for Windows (replacing the Xaml with API calls) and these samples run perfectly.
